test/parallel/test-web-locks.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+// Flags: --expose-gc
+
+require('../common');
+const { describe, it } = require('node:test');
+const assert = require('node:assert');
+const { Worker } = require('worker_threads');
+
+describe('Web Locks with worker threads', () => {
+  it('should handle exclusive locks', async () => {
+    const worker = new Worker(`
+      const { parentPort } = require('worker_threads');
+      const assert = require('node:assert');
+      
+      navigator.locks.request('exclusive-test', async (lock) => {
+        assert.strictEqual(lock.mode, 'exclusive');
+        parentPort.postMessage({ success: true });
+      }).catch(err => parentPort.postMessage({ error: err.message }));
```

Comment:
It doesn't necessarily need to be done in this PR, but we should make sure that this plays well with AsyncLocalStorage. Specifically, something like the following should work:

```
const als = new AsyncLocalStorage();
als.run(123, () => {
  navigator.locks.request('exclusive-test', async (lock) => {
    assert.strictEqual(als.getStore(), 123);
  });
});
```

This would suggest that the lock request is capable of capturing the current async context frame and restoring it when the lock has been acquired and the callback is invoked.